From 0e0828558c9d90004b5054a24ec24b34c9f21726 Mon Sep 17 00:00:00 2001
From: Xu Xuehui <xxh@rock-chips.com>
Date: Thu, 17 Aug 2017 18:57:56 +0800
Subject: [PATCH] net: wireless: rockchip_wlan: fix bcmdhd driver wowlan_triger
 error on android 6.0

stack backtrace:
[   11.444844] PC is at kfree+0xcc/0x250
[   11.445181] LR is at nl80211_set_wowlan+0x3e4/0x70c
[   11.445617] pc : [<ffffff80081a09e0>] lr : [<ffffff8008aa9414>] pstate: a0000145
[   11.446271] sp : ffffffc04308f7f0
[   11.446572] x29: ffffffc04308f7f0 x28: ffffffc06437f200
[   11.447061] x27: 0000000000000014 x26: 0000000000000000
[   11.447549] x25: ffffff80090b6000 x24: 0000000000000018
[   11.448036] x23: ffffffc072d00a40 x22: ffffff80090b6000
[   11.448522] x21: ffffff8008aa9414 x20: feeffffffffffffe
[   11.449008] x19: 0000000003ffffff x18: 0000000000000000
[   11.449494] x17: 0000007fb52815f4 x16: ffffff80088ee7ec
[   11.449981] x15: 0000000000000030 x14: 0000000000000014
[   11.450467] x13: 0000000000000005 x12: 0000000000000005
[   11.450953] x11: 0000000000000001 x10: 0000000000000001
[   11.451437] x9 : 0000000000000000 x8 : 0000000000000000
[   11.451922] x7 : 0000000000000000 x6 : ffffffc0437bb068
[   11.452408] x5 : 0000000000000040 x4 : 0000000000000000
[   11.452893] x3 : ffffffbebfffffc0 x2 : 0000000000000028
[   11.453380] x1 : 0000000000000000 x0 : 0000000000000000

Change-Id: Ibe1967413a35589a47a8ffe5e902f148afefa9ee
Signed-off-by: Xu Xuehui <xxh@rock-chips.com>
---
 drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/wl_cfg80211.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/wl_cfg80211.c b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/wl_cfg80211.c
index ea4b7a552883..dce81257732a 100644
--- a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/wl_cfg80211.c
+++ b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/wl_cfg80211.c
@@ -9586,6 +9586,9 @@ static s32 wl_setup_wiphy(struct wireless_dev *wdev, struct device *sdiofunc_dev
 		brcm_wowlan_config->patterns = NULL;
 		brcm_wowlan_config->n_patterns = 0;
 		brcm_wowlan_config->tcp = NULL;
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 19, 0))
+		brcm_wowlan_config->nd_config = NULL;
+#endif
 	} else {
 		WL_ERR(("Can not allocate memory for brcm_wowlan_config,"
 			" So wiphy->wowlan_config is set to NULL\n"));
-- 
2.35.3

